supervision/detection/utils.py

```diff
@@ -274,6 +275,81 @@ def box_non_max_suppression(
     return keep[sort_index.argsort()]
 
 
+def box_non_max_merge(
+    predictions: npt.NDArray[np.float64], iou_threshold: float = 0.5
+) -> Dict[int, List[int]]:
+    """
+    Apply greedy version of non-maximum merging to avoid detecting too many
+    overlapping bounding boxes for a given object.
+
+    Args:
+        predictions (npt.NDArray[np.float64]): An array of shape `(n, 5)` containing
+            the bounding boxes coordinates in format `[x1, y1, x2, y2]`
+            and the confidence scores.
+        iou_threshold (float, optional): The intersection-over-union threshold
+            to use for non-maximum suppression. Defaults to 0.5.
+
+    Returns:
+        Dict[int, List[int]]: Mapping from prediction indices
+        to keep to a list of prediction indices to be merged.
+    """
+    keep_to_merge_list: Dict[int, List[int]] = {}
+
+    scores = predictions[:, 4]
+    order = scores.argsort()
+
+    while len(order) > 0:
+        idx = order[-1]
+        merge_candidate = np.expand_dims(predictions[idx], axis=0)
+
+        order = order[:-1]
+        if len(order) == 0:
+            keep_to_merge_list[idx.tolist()] = []
+            break
+
+        ious = box_iou_batch(predictions[order][:, :4], merge_candidate[:, :4])
+        ious = ious.flatten()
+
+        above_threshold = ious >= iou_threshold
+        keep_to_merge_list[idx] = np.flip(order[above_threshold]).tolist()
+        order = order[~above_threshold]
+
+    return keep_to_merge_list
+
+
+def box_non_max_merge_batch(
+    predictions: npt.NDArray[np.float64], iou_threshold: float = 0.5
```

Comment:
Let's mimic the behavior of `mask_non_max_suppression` and `box_non_max_suppression`. If predictions is `(n, 6)`, shape `box_non_max_merge_batch` is class-based, and if it is `(n, 5)`, it is class agnostic.